# **Week 10: Large Sample Hypothesis Testing**

```
.------------------------------------.
|   __  ____  ______  _  ___ _____   |
|  |  \/  \ \/ / __ )/ |/ _ \___  |  |
|  | |\/| |\  /|  _ \| | | | | / /   |
|  | |  | |/  \| |_) | | |_| |/ /    |
|  |_|  |_/_/\_\____/|_|\___//_/     |
'------------------------------------'

---

Through the following examples, we will explore the concepts of (large-sample) hypothesis testing (LSHT)
and examine their practical implications.

## **Pre-Configurating the Notebook**

### **Switching to the R Kernel on Colab**

By default, Google Colab uses Python as its programming language. To use R instead, you’ll need to manually switch the kernel by going to **Runtime > Change runtime type**, and selecting R as the kernel. This allows you to run R code in the Colab environment.

However, our notebook is already configured to use R by default. Unless something goes wrong, you shouldn’t need to manually change runtime type.

### **Importing Required Packages**
**Run the following lines of code**:

In [ ]:
#Do not modify

setwd("/content")

# Remove `MXB107-Notebooks` if exists,
if (dir.exists("MXB107-Notebooks")) {
  system("rm -rf MXB107-Notebooks")
}

# Fork the repository
system("git clone https://github.com/edelweiss611428/MXB107-Notebooks.git")

# Change working directory to "MXB107-Notebooks"
setwd("MXB107-Notebooks")

#
invisible(source("R/preConfigurated.R"))

**Do not modify the following**

In [ ]:
if (!require("testthat")) install.packages("testthat"); library("testthat")

test_that("Test if all packages have been loaded", {

  expect_true(all(c("ggplot2", "tidyr", "dplyr", "stringr", "magrittr", "knitr") %in% loadedNamespaces()))

})

### **Demonstrating Example**

A phone company advertises that the average battery life of their phones (when continuously watching videos), denoted as $\mu$, is 8 hours.

To verify this claim, an independent random sample of 31 phones was tested.

In [6]:
smpl_data = c(7.2, 8.53, 8.07, 7.99, 7.79, 7.77, 8.9, 7.64, 7.35, 8.45, 9.14, 7.93, 7.35, 7.52, 7.41, 8.27,
7.55, 7.5, 8.53, 8.37, 8.17, 8.15, 8.02, 7.63,7.64, 8.83, 8.17, 7.41, 7.7, 8.21, 7.13)

[1] 31

**Write down the null and alternative hypotheses for testing whether the company’s claim is correct.**


Since we are testing whether or not there is evidence *against* the company’s claim that the average battery life is 8 hours, the alternative hypothesis should challenge this claim.  

Because we are not concerned if the battery lasts longer than 8 hours (that would be favorable to consumers), we only test if it is **less** than 8 hours.  

$$
\begin{align}
H_0: \mu &= 8 \\
H_1: \mu &< 8
\end{align}
$$

This is a **left-tailed test** (or often simply referred to as one-tailed or one-sided test) of the mean.

**Approximate the sampling distribution of the sample mean under the null hypothesis.**

Since our sample size is larger than 30, it is reasonable to assume the sample mean approximately has a sampling distribution that is normal.  We do not know the population variance of battery life, but we can estimate it from the sample.

In [8]:
var(smpl_data)
sd(smpl_data)

length(smpl_data)

mean(smpl_data)


[1] 0.2673452

[1] 0.5170543

[1] 31

[1] 7.945806

$$\bar{x} \sim \mathcal{N}\Big(7.95,\frac{0.267}{31}\Big)$$

**Define the z-test statistic for testing the null hypothesis**


$$
Z = \frac{\bar{x} - \mu_0}{\frac{s}{\sqrt{n}}} \approx \frac{7.95 - 8}{\frac{0.517}{\sqrt{31}}} \approx -0.538
$$


**Now we want to compute the p-value**

Since we have a left-tailed test, the p-value is given by
$$
P(Z < -0.538 | H_0 \mbox{ is true})
$$

This is the probability of getting our observed test statistic value or more extreme (in the direction of the alternative) given that the null is true.  If this probability is small, it would provide evidence against the null in favour of the alternative.

Since $Z$ is approximately standard normal, we can compute the p-value using the cdf of the standard normal distribution (we need R to help us do that).  The full set of R code to conduct the hypothesis test is below.  

In [9]:
xbar = mean(smpl_data)
s = sd(smpl_data)
n = 31
# compute the test statistic
Z = (xbar-8)/(s/sqrt(n))
Z

p = pnorm(Z)
p


[1] -0.5835691

[1] 0.2797551

Here we see the p-value is quite large, around 0.28. Thus it was reasonably likely to get the sample we got (or more extreme) if the null was true.  Thus there is insufficient evidence against the null.  That is, from this data, there is no evidence against the assumption that the mean battery life is 8 hours.

## **Workshop Questions**




### **Question 1**

The following questions are based on the `episodes` dataset. While you are expected to use R to compute the answers, the underlying concepts are identical to those in pen-and-paper hypothesis testing calculations.

In [ ]:
episodes = read.csv("./datasets/episodes.csv")
episodes %>% str()

'data.frame':	704 obs. of  57 variables:
 $ Series                        : chr  "TOS" "TOS" "TOS" "TOS" ...
 $ Series.Name                   : chr  "The Original Series" "The Original Series" "The Original Series" "The Original Series" ...
 $ Season                        : int  1 1 1 1 1 1 1 1 1 1 ...
 $ Episode                       : int  1 2 3 4 5 6 7 8 9 10 ...
 $ IMDB.Ranking                  : num  7.3 7.2 7.8 8 7.8 6.9 7.6 7.1 7.5 8.2 ...
 $ Title                         : chr  "The Man Trap" "Charlie X" "Where No Man Has Gone Before" "The Naked Time" ...
 $ Star.date                     : chr  "1513.1" "1533.6" "1312.4" "1704.2" ...
 $ Air.date                      : chr  "8/9/66" "15/9/66" "22/9/66" "29/9/66" ...
 $ Bechdel.Wallace.Test          : logi  FALSE FALSE FALSE FALSE FALSE FALSE ...
 $ Director                      : chr  "Marc Daniels" "Lawrence Dobkin" "James Goldstone" "Marc Daniels" ...
 $ Writer.1                      : chr  "George Clayton Johnson" "Gene Rode

#### **Question 1.1**

Is there evidence that the average IMDB rating of Star Trek: The Original Series episodes is lower than 7.7? Interpret the results for a non-statistician stakeholder.

In [ ]:
#H0: mu = 7.7
#H1: mu < 7.7

episodes %>%
  filter(Series == "TOS") %>%
  summarise(avgIMDB = mean(IMDB.Ranking),
            sdIMDB = sd(IMDB.Ranking),
            n = n(),
            mu_H0 = 7.7,
            z_05 = qnorm(0.05)) %>%
  mutate(z_obs = (avgIMDB - mu_H0)/(sdIMDB/sqrt(n)))

#As z_obs < z_05, at 95% significance level, there is evidence against the null hypothesis that the average IMDB ranking of TOS is 7.7.
# We reject the null hypothesis in favour of the alternative one that the average rating of TOS is lower than 7.7


avgIMDB,sdIMDB,n,mu_H0,z_05,z_obs
<dbl>,<dbl>,<int>,<dbl>,<dbl>,<dbl>
7.53625,0.8038269,80,7.7,-1.644854,-1.822065


#### **Question 1.2**


Is there evidence that the mean IMDB rating of episodes of Star Trek: The Original Series differs from the mean rating of Star Trek: The Next Generation? Interpret the results for a non-statistician stakeholder.

In [ ]:
#H0: mu_TOS - mu_TNG = 0
#H1: mu_TOS - mu_TNG != 0

episodes %>%
  filter(Series %in%  c("TOS", "TNG")) %>%
  group_by(Series) %>%
  summarise(avgIMDB = mean(IMDB.Ranking),
            varIMDB = var(IMDB.Ranking),
            n = n()) -> summaryStats

TNG_stats = summaryStats %>% filter(Series == "TNG")
TOS_stats = summaryStats %>% filter(Series == "TOS")

z_obs = (TOS_stats$avgIMDB - TNG_stats$avgIMDB)/sqrt(TOS_stats$varIMDB/TOS_stats$n + TNG_stats$varIMDB/TNG_stats$n)
z_975 = qnorm(0.975)

abs(z_obs) > z_975

#As abc(z_obs) < z_975, at 95% significance level, there is no evidence against the null hypothesis that
#the average ranking of TOS is equal to that of TNG. Do not reject the null hypothesis.

[1] FALSE

#### **Question 1.3**

Is there evidence that the proportion of Star Trek: The Next Generation episodes that pass the Bechdel-Wallace Test is different from 0.4? Interpret the results for a non-statistician stakeholder.


**Note**: While these series have ended and you technically have the full “population” of some values (e.g., results of the Bechdel-Wallace test), we still ask you to test whether or not the proportion of episodes that pass the test is equal to 0.4. This may seem counter-intuitive, but you can think of it as follows:

- The episode test results are treated as realisations from an unknown probability distribution $f$ (here, Bernoulli(p)).
- Although the episodes are released, we are interested in the underlying process that generates these values. This includes not-yet-released episodes or hypothetical similar episodes. Simply examining the “complete” population of Bechdel test results is not sufficient; instead, we rely on a statistical model to quantify uncertainty.

In [ ]:
#H0: p = 0.4
#H1: p != 0.4

episodes %>%
  filter(Series == "TNG") %>%
  summarise(p_hat = mean(Bechdel.Wallace.Test),
            n = n(),
            p_H0 = 0.4,
            z_975 = qnorm(0.975)) %>%
  mutate(z_obs = abs((p_hat-p_H0)/sqrt(p_H0*(1-p_H0)/n)))


#As abc(z_obs) < z_975, at 95% significance level, there is no evidence against the null hypothesis that
#the proportion of TNG episodes that pass the Bechdel-Wallace Test is 0.4. Do not reject the null hypothesis.

p_hat,n,p_H0,z_975,z_obs
<dbl>,<int>,<dbl>,<dbl>,<dbl>
0.4382022,178,0.4,1.959964,1.040383


#### **Question 1.4**


Is there evidence that the proportion of episodes that pass the Bechdel-Wallace Test differs between Star Trek: The Next Generation and Star Trek: Voyager? Interpret the results for a non-statistician stakeholder.

In [ ]:
#H0: p_TOS - p_TNG = 0
#H1: p_TOS - p_TNG != 0

episodes %>%
  filter(Series %in%  c("TOS", "TNG")) %>%
  group_by(Series) %>%
  summarise(p_hat = mean(Bechdel.Wallace.Test),
            n = n()) -> summaryStats

TNG_stats = summaryStats %>% filter(Series == "TNG")
TOS_stats = summaryStats %>% filter(Series == "TOS")

z_obs = (TOS_stats$p_hat - TNG_stats$p_hat)/sqrt(TOS_stats$p_hat*(1-TOS_stats$p_hat)/TOS_stats$n + TNG_stats$p_hat*(1-TNG_stats$p_hat)/TNG_stats$n)
z_975 = qnorm(0.975)

abs(z_obs) > z_975

#As abc(z_obs) > z_975, at 95% significance level, there is evidence against the null hypothesis that
#the proportion of episodes passing the Bechdel test of TOS is equal to that of TNG. Reject the null hypothesis in favour of the
#alternative one that these two proportions are different.

[1] TRUE

### **Question 2**

The following questions are based on the `epa_data` dataset. While you are expected to use R to compute the answers, the underlying concepts are identical to those in pen-and-paper hypothesis testing calculations.


In [ ]:
epa_data = read.csv("./datasets/epa_data.csv")
epa_data %>% str()

'data.frame':	13569 obs. of  9 variables:
 $ city : int  16 15 16 19 19 19 19 19 19 19 ...
 $ hwy  : int  24 22 22 27 29 24 26 27 29 24 ...
 $ cyl  : int  8 8 8 4 4 4 4 4 4 4 ...
 $ disp : num  5 5 5 2 2 2.4 2.4 2 2 2.4 ...
 $ drive: chr  "Rear-Wheel Drive" "Rear-Wheel Drive" "Rear-Wheel Drive" "Rear-Wheel Drive" ...
 $ make : chr  "Jaguar" "Jaguar" "Jaguar" "Pontiac" ...
 $ model: chr  "XK" "XK" "XK Convertible" "Solstice" ...
 $ trans: chr  "Automatic" "Automatic" "Automatic" "Automatic" ...
 $ year : int  2010 2010 2010 2010 2010 2010 2010 2010 2010 2010 ...


#### **Question 2.1**

Is there evidence of a difference in the average city mileage between cars manufactured in 2015 and 2020?  Interpret the results for a non-statistician stakeholder.

In [ ]:
#H0: mu_2020 - mu_2015 = 0
#H1: mu_2020 - mu_2015 != 0

epa_data %>%
  filter(year %in%  c(2015, 2020)) %>%
  group_by(year) %>%
  summarise(avgMileage = mean(city),
            varMileage = var(city),
            n = n()) -> summaryStats

epa_2015 = summaryStats %>% filter(year == 2015)
epa_2020 = summaryStats %>% filter(year == 2020)

z_obs = (epa_2020$avgMileage - epa_2015$avgMileage)/sqrt(epa_2020$varMileage/epa_2020$n + epa_2015$varMileage/epa_2015$n)
z_975 = qnorm(0.975)

abs(z_obs) > z_975

#As abc(z_obs) > z_975, at 95% significance level, there is evidence against the null hypothesis that
#there is no difference between the average city milages between cars manufactured of TOS in 2015 and 2020. Reject the null hypothesis.

[1] TRUE



<details>
<summary>▶️ Click to show the solution</summary>

Solution will be released at the end of the week!

</details>


#### **Question 2.2**

Is there evidence that the proportion of cars produced with manual transmissions in 2010 is greater than 0.5? Interpret the results for a non-statistician stakeholder.

In [ ]:
#H0: p = 0.5
#H1: p > 0.5


epa_data %>%
  filter(year == 2010) %>%
  summarise(p_hat = mean(ifelse(trans == "Manual", 1, 0)),
            n = n(),
            p_H0 = 0.5,
            z_95 = qnorm(0.95)) %>%
  mutate(z_obs = (p_hat-p_H0)/sqrt(p_H0*(1-p_H0)/n))

#As z_obs < z_95, at 95% significance level, there is no evidence against the null hypothesis that
#the proportion of cars produced with manual transmissions in 2010 is 0.5. Do not reject the null hypothesis.

p_hat,n,p_H0,z_95,z_obs
<dbl>,<int>,<dbl>,<dbl>,<dbl>
0.2434626,1109,0.5,1.644854,-17.08624


#### **Question 2.3**

Is there evidence that the proportion of cars produced with manual transmissions for the years 2011 and 2012 has decreased? Interpret the results for a non-statistician stakeholder.

In [ ]:
#H0: mu_2012 - mu_2011 = 0
#H1: mu_2012 - mu_2011 < 0

epa_data %>%
  filter(year %in%  c(2012, 2011)) %>%
  group_by(year) %>%
  summarise(p_hat = mean(ifelse(trans == "Manual", 1, 0)),
            n = n()) -> summaryStats

epa_2012 = summaryStats %>% filter(year == 2012)
epa_2011 = summaryStats %>% filter(year == 2011)

z_obs = (epa_2012$p_hat - epa_2011$p_hat)/sqrt(epa_2012$p_hat*(1-epa_2012$p_hat)/epa_2012$n + epa_2011$p_hat*(1-epa_2011$p_hat)/epa_2011$n)
z_05 = qnorm(0.05)

z_obs < z_05

#As z_obs > z_05, at 95% significance level, there is no evidence against the null hypothesis that the proportion of cars
# produced with manual transmissions for the years 2011 and 2012 has remained the same. Do not reject the null hypothesis.


[1] FALSE